src/wallet/interfaces.cpp

```diff
@@ -549,6 +549,14 @@ class WalletClientImpl : public WalletClient
         options.require_existing = true;
         return MakeWallet(m_context, LoadWallet(m_context, name, true /* load_on_start */, options, status, error, warnings));
     }
+    std::unique_ptr<Wallet> restoreWallet(const std::string& backup_file, const std::string& wallet_name, bilingual_str& error, std::vector<bilingual_str>& warnings) override
```

Comment:
This PR can't make changes outside of the qt directory. You can make these changes in the core repo and then base this PR on that.

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/23721